src/java.net.http/share/classes/jdk/internal/net/http/common/ValidatingHeadersConsumer.java

```diff
@@ -26,15 +26,36 @@
 
 import java.io.IOException;
 import java.io.UncheckedIOException;
+import java.util.Map;
 import java.util.Set;
 
 /*
  * Checks RFC 9113 rules (relaxed) compliance regarding pseudo-headers.
  */
 public class ValidatingHeadersConsumer {
 
-    private static final Set<String> PSEUDO_HEADERS =
-            Set.of(":authority", ":method", ":path", ":scheme", ":status");
+    private final Context context;
+
+    public ValidatingHeadersConsumer(Context context) {
+        this.context = context;
+    }
+
+    public enum Context {
+        REQUEST,
+        RESPONSE,
+        TRAILER
```

Comment:
Do we need `TRAILER`, given we don't use it, AFAIU?

Reply:
Removed